### ISSUE TITLE
Feature request: support for other input files

### ISSUE THREAD

**Issue by din14970** [CDoberstein/ExitWaveReconstruction#2]
We collect our experimental data mostly on FEI Titan instruments and the focal series are stored as individual SER/EMI files. I can convert these files into individual TIFFS, but I can not combine them and save them out to a DM3 file. Would it be possible to make it so that `input file` also accepts a folder with tiffs or some other arrangement?

**Comment by CDoberstein** [comment]
Yes, it should be easily possible to extend the program to accept other formats such as tiff or png. I will extend the source code and get back to you later.

By the way, the program is already able to read a directory containing multiple dm3 or q2bz files, each of which contains a single TEM image. This option is not documented in the template parameter file, but instead is a part of the documentation of the function `LoadInputImageSeries` in the file [LoadUtilities.h](https://github.com/CDoberstein/ExitWaveReconstruction/blob/master/Src/projects/ExitWaveReconstruction/LoadUtilities.h#L89). Basically, you can set the parameter `inputDataFile` to point to a nonexisting file "abc.dm3" and then the program will search for individual .dm3 or .q2bz image files in the directory "abc/". This way of providing input files to the program is intended to be used only when a previously stopped reconstruction is resumed, but it may provide a nice workaround for now.

I uploaded additional conversion functions to the [tools/image/converter](https://github.com/CDoberstein/ExitWaveReconstruction/tree/master/Src/tools/image/converter) directory that can be used to convert SER, TIFF, CImg and DM3 files to the Quocmesh .q2bz format. This way you should be able to test the exit wave reconstruction on your input data until I've extended it to read other formats directly. For the conversion of SER and CImg files you'll need the cimg library and have to set `-DUSE_CIMG=1` in `goLinux.sh`.

**Comment by din14970** [comment]
Thanks, this might indeed be a nice workaround! Thanks for updating the repo and the detailed explanations. I succesfully rebuilt it on my personal Linux laptop. If you add a new minor version I can try to update the recipe on conda-forge. The converter also seems to work, though I guess it only accepts one image at a time? Will try to convert my images and run a test reconstruction then get back to you. Once it works I can try to make a python wrapper which I'm sure our group and maybe other microscopy groups will find very helpful. Will keep you in the loop if you are interested.

**Comment by CDoberstein** [comment]
I just uploaded a new version of the code that adds support for reading individual dm3, q2bz, tiff, ser and png files from a directory and added a new minor version release. I didn't test the code for reading ser files, but if the converter worked for you, the new version of the code should work as expected.

Some of the converters unfortunately only convert one image at a time. If you want to convert many images you could consider writing a shell script that executes the converter for each image automatically.

Please do keep me in the loop! I am very interested to know if you can make good use of the software.

**Comment by din14970** [comment]
As mentioned in my e-mail, the converter SER -> quoc works nicely but it adds a 000 to all the filenames. So running `Reconstruction` on a folder of SER file unfortunately doesn't work out of the box. As a workaround I run the following shell script to convert the SER files to quoc and put those files with the same name back in the same folder; then `Reconstruction` works fine when run on a folder of quoc files. My SER files are already renamed to ImageXXX using python, but I might change the shell script to force the quoc files to be named correctly.

```
#!/bin/bash
DIREC="$*"
echo "Converting SER files in $DIREC"
for i in "${DIREC}"*.ser
do
    echo "Found $i"
    [ -f "$i" ] || continue
    convertSERToQuoc "${i}"
    filename=$(basename -- "$i")
    filename="${filename%.*}"
    quocname="${filename}000.q2bz"
    newname="${filename}.q2bz"
    echo "Moving $quocname to $DIREC$newname"
    [ -f "$quocname" ] || continue
    mv "$quocname" "${DIREC}${newname}" 
    rm "$i"
    echo "Removed $i"
    echo "------------------"
done
```

**Comment by CDoberstein** [comment]
I just updated the code so that no trailing "000" should be appended to the filenames of the q2bz files anymore. Please try to run the reconstruction with the SER files again to check if the problem is solved.

(I only changed the code for reading and converting SER files with the `Reconstruction` executable. The behavior of `tools/image/converter/convertSERToQuoc` is still the same.)
